test/e2e/storage/external/external.go

```diff
@@ -304,17 +317,56 @@ func (d *driverDefinition) GetDynamicProvisionStorageClass(config *testsuites.Pe
 	return testsuites.GetStorageClass(sc.Provisioner, sc.Parameters, sc.VolumeBindingMode, f.Namespace.Name, "e2e-sc")
 }
 
+func loadSnapshotClass(filename string) (*unstructured.Unstructured, error) {
+	data, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return nil, err
+	}
+	snapshotClass := &unstructured.Unstructured{}
+
+	if err := runtime.DecodeInto(scheme.Codecs.UniversalDecoder(), data, snapshotClass); err != nil {
+		return nil, errors.Wrap(err, filename)
+	}
+
+	return snapshotClass, nil
+}
+
 func (d *driverDefinition) GetSnapshotClass(config *testsuites.PerTestConfig) *unstructured.Unstructured {
-	if !d.SnapshotClass.FromName {
+	if !d.SnapshotClass.FromName && d.SnapshotClass.FromFile == "" && d.SnapshotClass.FromExistingClassName == "" {
 		e2eskipper.Skipf("Driver %q does not support snapshotting - skipping", d.DriverInfo.Name)
 	}
 
-	snapshotter := d.DriverInfo.Name
-	parameters := map[string]string{}
-	ns := config.Framework.Namespace.Name
-	suffix := snapshotter + "-vsc"
+	var snapshotClass *unstructured.Unstructured
+	var err error
+	f := config.Framework
+
+	switch {
+	case d.SnapshotClass.FromName:
+		snapshotter := d.DriverInfo.Name
+		parameters := map[string]string{}
+		ns := config.Framework.Namespace.Name
```

Comment:
Can we have all 3 cases use `GetSnapshotClass` so that they all use the same ns and suffix formats?

Reply:
I also thought that it would be nice to do so, after I wrote my last comment.
However, to do so, we need to get `parameters` from `*unstructured.Unstructured`. Is there a good way to achieve it? I couldn't find a good way from [here](https://godoc.org/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured#Unstructured).

Reply:
Does something like [this](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/storage/testsuites/snapshottable.go#L199) work?

Reply:
Fixed and rebased. PTAL